http/middleware/log_request.go

```diff
@@ -2,44 +2,167 @@ package middleware
-// using the enclosed implementation of logger.Logger.
+const (
+	passwordParam     = "password"
+	contentLenHeader  = "Content-Length"
+	contentTypeHeader = "Content-Type"
+	referrerHeader    = "Referrer"
+	userAgentHeader   = "User-Agent"
+)
+
+// LogRequest logs the a LogRequestRecord using the provided handler.
 //
-// LogRequest scrubs the values for the following keys:
+// For the LogRequestRecord.URI, LogRequest masks query params matching these keys with trails.LogMaskVal:
 // - password
 //
-// if logger.Logger is nil, NoopAdapter returns and this middleware does nothing.
-func LogRequest(ls logger.Logger) Adapter {
-	if ls == nil {
+// If handler is nil, NoopAdapter returns and this middleware does nothing.
+func LogRequest(handler slog.Handler) Adapter {
+	if handler == nil {
 		return NoopAdapter
 	}
 
+	// NOTE(dlk): add constant values to include on every message here
+	ls := slog.New(handler.WithAttrs([]slog.Attr{
+		{Key: trails.LogKindKey, Value: trails.HTTPLogKind},
+	}))
+
 	return func(h http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			uri := r.URL.Path
-			q := r.URL.Query()
-			if val := q.Get("password"); val != "" {
-				q.Set("password", "xxxxxxx")
-			}
-
-			query := q.Encode()
-			if query != "" {
-				uri += "?" + q.Encode()
-			}
-
-			strs := []string{r.Method, uri}
-			val := r.Context().Value(IpAddrCtxKey)
-			if val != nil {
-				strs = append([]string{val.(string)}, strs...)
-			}
-
-			ls.Info(strings.Join(strs, " "), nil)
-			h.ServeHTTP(w, r)
+			start := time.Now()
+
+			writer := &requestLogger{ResponseWriter: w, status: http.StatusOK}
+			h.ServeHTTP(writer, r)
+
+			end := time.Since(start).Milliseconds()
+
+			rec := newRecord(writer, r)
+			rec.Duration = end
+
+			var msg string // NOTE(dlk): no message for now.
+
+			// NOTE(dlk): LogAttrs is not the standard method to use,
+			// but we know we only have a []slog.Attr to report,
+			// so let's leverage it.
+			//
+			// TODO(dlk): using r.Context() may not be appropriate.
+			// Unfortunately, outside of configuring a slog.Handler to pull
+			// values from a context.Context into slog.Attr,
+			// contexts appear to not be used by slog.
+			// This includes not watching for cancellations.
+			// The ranger.Context may be the best fit for this.
+			// We'll reassess when slog makes into a Go version.
+			ls.LogAttrs(r.Context(), slog.LevelInfo, msg, rec.attrs()...)
 		})
 	}
 }
+
+// A LogRequestRecord represents the fields that a LogRequest
+type LogRequestRecord struct {
+	BodySize       int    `json:"responseBodySize"`
+	Duration       int64  `json:"duration"`
+	Host           string `json:"host"`
+	ID             string `json:"id"`
+	IPAddr         string `json:"remoteAddr"`
+	Method         string `json:"method"`
+	Path           string `json:"path"`
+	Protocol       string `json:"protocol"`
+	Referrer       string `json:"referrer"`
+	ReqContentLen  int    `json:"requestContentLength"`
```

Comment:
IMO since we're omitting `request` on the names of the other request params (and it's implied) then these should just be `contentLength` and `contentType`

Reply:
Glad you bring this up. What's your appetite on using `slog`'s grouping functionality? Cf., [`slog.Group`](https://pkg.go.dev/golang.org/x/exp/slog#Group), which has a plain text example, rewritten as JSON, those `req` fields would be an object:
```json
"req": { "method": "GET", "url": "localhost" }
```
Personally, I'd rather go this direction, but comes down to whether traversing these nested relationships is a PITA for CloudWatch or whether it's simply easier to keep it flat.

Reply:
I went ahead with your suggestions, but it's worth a gander at `slog.Group`. It's a quick change if you like the way it formats logs.

Reply:
honestly/unfortunately it's going to be hard for me to know for sure until I get some stuff piped into cloudwatch to get a better sense of what the ergonomics are like. I can tell you right now that I prefer having to type out `contentType` a whole lot more than having to type out `requestContentType` or `{"req": {"contentType": ...}}`, but maybe once I get into cloudwatch I'll see that its magic 🪄 makes all of this super easy no matter how we've formatted it.

I say we go with what you have here now (flat & terse), and just extend me some grace to maybe want to take a 2nd pass at this once I get CW dashboards wired up 🤞 🙏